tests/unit/s3middleware/LifecycleUtils.spec.js

```diff
@@ -90,6 +90,37 @@ describe('LifecycleUtils::getApplicableRules', () => {
             res.NoncurrentVersionExpiration.NoncurrentDays, 3);
     });
 
+    it('should return earliest applicable rules', () => {
```

Comment:
Is the above test still bringing value, considering this new test? Maybe we should remove it, otherwise setting their name differently could help distinguishing them.